### DISCUSSION TITLE
Creating related rows in more than one table

### DISCUSSION THREAD

**Discussion by doingandlearning** [Budibase/budibase#4998] [Help] [answered]
Hey :)

When I create a form, I can set it's type and schema. For what I'm trying to do I'd assume I need to set up a custom schema but can't seem to work out how to do it.

I have three tables:
- Slots
- Leaders
- Booking

A booking has a 1->many relationship with the other two tables. I'd like a form that will allow me to create a new leader (preferably in-line). So, submitting the form would create the leader record and link it to that field in the booking.

Is this possible? 

Thanks,
Kevin.

**Comment by melohagan** [comment, marked as answer]
Hey Kevin!

You can create multiple linked entries in one form as follows:

 In the _/booking/new/row_ screen, delete the drop down for the leader, and replace it with a new text field. 
Type in the field name: **LeaderName**

![Screenshot 2022-03-22 at 12 01 01](https://user-images.githubusercontent.com/101575380/159480786-8c100a27-d798-412d-bbfc-05d00ce718f5.png)

Next click on the 'Save' button and 'Define actions'

Here you will need to add another Save Row action underneath the existing one for the Leader table. This will allow you to create a new row for the leader entered into the in-line **LeaderName** field. 

![Screenshot 2022-03-22 at 12 17 06](https://user-images.githubusercontent.com/101575380/159481463-854513b8-7fcc-44da-a0ba-d75ffcfa35fe.png)

Two columns overrides must be added. The first **Name** `{{ Form.Fields.LeaderName}}` saves the leader name as the value from our new **LeaderName** form field. 

The second **Booking** column (the relationship field) is set to `{{ Action 2.Saved row._id }}`. This represents the relationship id of the booking row that was saved in the previous action. 

Hopefully this helps!

**Reply by doingandlearning** [reply]
This is super helpful!

Two further things:
- the booking column though seems to be expecting a 1-1 relationship - is there a way to append this booking in a many-to-one way? 

- Secondly, is this documented anywhere but here? Being able to reference the output of previous actions is super cool, just wondering if there is a more comprehensive docs than https://docs.budibase.com/docs/actions here?


Cheers!

**Reply by melohagan** [reply]
Glad to help :)

The Save Row action can only save one row at a time, so if you wanted to save a fixed number of leaders against one new booking, you would need separate text fields and actions for each new leader. 

There should be a release by the end of April to support for-each blocks in automations. This would allow you to have a Row Created trigger on your bookings table, that would then loop over an array of leaders (saved from your form into a JSON column) and then Create Row for each entry into the leaders table.

As for the documentation, I'll update it to include this information about referencing previous actions.

**Comment by Markus-** [comment]
How would I put both inserts into a transaction? Is this possible without an individual query?
